spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/security/oauth2/OAuth2Properties.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2012-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.security.oauth2;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.security.oauth2.client.registration.ClientRegistrationProperties;
+
+/**
+ * Configuration properties for Spring Security OAuth 2.0 / OpenID Connect 1.0 support.
+ *
+ * @author Joe Grandja
+ * @since 2.0.0
+ */
+@ConfigurationProperties(prefix = "security.oauth2")
+public class OAuth2Properties {
+
+	private Map<String, ClientRegistrationProperties> clients = new LinkedHashMap<>();
```

Comment:
I am not sure I like the idea that we bind a 3rd party class directly to the environment. We've done that in the past and regretted it. It's just a raw observation though, maybe that's fine.

Reply:
Let me know if you would like a new class similar to `ClientRegistrationProperties` in Boot. Either way, we will keep `ClientRegistrationProperties` in Spring Security as it serves as a convenience class in certain use cases.

Reply:
I think the defaults and how they are applied should largely derive from that. I can certainly appreciate you have an infrastructure in the project that we would reuse. I just don't think binding it directly to the environment is a good idea.

Reply:
Can you elaborate on a specific issue have you had in the past where you used a 3rd party class to bind in the `Environment`? I would like to understand this more.